src/server/api/AdminPanel.ts

```diff
@@ -0,0 +1,208 @@
+import type { CustomRequest } from "../../types.js";
+import type { Response } from "express";
+// @ts-ignore
+import { deleteUser, getMemberDataByCriteria, updateMemberColumns } from "../database/memberManager.js";
+// @ts-ignore
+import { deleteRefreshTokensOfUser } from "../database/refreshTokenManager.js";
+// @ts-ignore
+import { logEvents } from "../middleware/logEvents.js";
+// @ts-ignore
+import { deleteAccount } from "../controllers/deleteAccountController.js";
+// @ts-ignore
+import { deleteAllSessionsOfUser } from "../controllers/authenticationTokens/sessionManager.js";
+
+const validCommands = [
+	"ban",
+	"delete",
+	"username",
+	"logout",
+	"verify",
+	"post",
+	"invites",
+	"announce",
+	"userinfo",
+	"help"
+];
+
+function processCommand(req: CustomRequest, res: Response): void {
+	const command = req.params["command"]!;
+	const commandAndArgs = command.split(" ");
+	if (!req.memberInfo.signedIn) {
+		res.status(401).send("Cannot send commands while logged out.");
+		return;
+	}
+	if (!(req.memberInfo.roles?.includes("admin") ?? false)) {
+		res.status(403).send("Cannot send commands without the admin role");
+		return;
+	}
+	// TODO prevent affecting accounts with equal or higher roles
+	switch (commandAndArgs[0]) {
+		case "ban":
+			return;
+		case "delete":
+			deleteCommand(command, commandAndArgs, res);
+			return;
+		case "username":
+			usernameCommand(command, commandAndArgs, res);
+			return;
+		case "logout":
+			logoutUser(command, commandAndArgs, res);
+			return;
+		case "verify":
+			return;
+		case "post":
+			return;
+		case "invites":
+			return;
+		case "announce":
+			return;
+		case "userinfo":
+			getUserInfo(command, commandAndArgs, res);
+			return;
+		case "help":
+			helpCommand(commandAndArgs, res);
+			return;
+		default:
+			res.status(422).send("Unknown command.");
+			return;
+	}
+}
+
+function deleteCommand(command: string, commandAndArgs: string[], res: Response) {
+	logCommand(command);
+	if (commandAndArgs.length < 2) {
+		res.status(422).send("Invalid number of arguments, expected 1, got " + (commandAndArgs.length - 1) + ".");
+		return;
+	}
+	const reason = commandAndArgs[2];
+	const username = commandAndArgs[1];
+	const userid = getMemberDataByCriteria(["user_id"], "username", username)["user_id"];
+	if (userid !== undefined) {
+		deleteAccount(userid, reason ?? "");
```

Comment:
deleteAccount() returns a boolean whether the deletion was a success or not. It could be confusing to the admin if they get a success message when it actually fails. If this is false, send the admin a failure message instead (the function does not return why it failed).